waku/node/rest/lightpush/types.nim

```diff
@@ -0,0 +1,65 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  std/[sets, strformat],
+  chronicles,
+  json_serialization,
+  json_serialization/std/options,
+  presto/[route, client, common]
+
+import
+  ../../../common/base64,
+  ../../../waku_core,
+  ../relay/types as relay_types,
+  ../serdes
+
+export relay_types
+
+#### Types
+
+type PushRequest* = object
+      pubsubTopic*: Option[PubSubTopic]
```

Comment:
Per the RFC the topic is not optional but I guess we are going to update the RFC soon :tm:

Reply:
Well, it was not obvious and I followed node interface lightpushPublish which has as optional due to sharding. Isn't it?

Reply:
Yeah and one more. I'm not pretty sure if it is ok to push a message with empty pubsubTopic and empty contentTopic, that seems doable now. I'm actually testing it further meanwhile.

Reply:
Ok maybe I was confused is `PushRequest` used for client -> server or is it between rest -> node?

client -> server; should follow the RFC
rest -> node; should allow optional pubsub with mandatory content topic

Reply:
Exactly the latter. You ask your node (presumably a light node) to make a push request toward a service provider relay node. So as in this case sharding will take place in case no pubsubTopic provided.